### ISSUE TITLE
Install failed on RetroPie image 4.2

### ISSUE THREAD

**Issue by caver01** [mholgatem/gpioneer#5]
Imaged an SD card with RetroPie 4.2. Booted with a Pi 3 and installed gpioneer per instructions. Install failed--"Invalid operation python-pip"

I tried installing EVDEV package etc. (will try to get the exact apt-get command I used) on a whim and re-ran the gpioneer install. it worked!

This should be fairly easy to reproduce. I think RetroPie 4.2 is based on Raspbian Jessie-lite.

**Comment by mholgatem** [comment]
I found the error in the install script and fixed it. 

I wrote:

> `sudo apt-get python pip`

rather than:
> `sudo apt-get -y install python pip`

On a side note, I'm going to buy a Pi 3 tomorrow and will finally get around to fixing that other bug. I also plan to add some extra functionality like the ability to create multiple controllers and the like. Let me know if there's a feature that you've been hankering for.

**Comment by caver01** [comment]
Great! I have been commenting that one line in the script (workaround mentioned in the other issue) to overcome the slow restart/shutdown bug, but for the most part, I really like gpioneer. I must say, that I have also mentioned it a few times on the RetroPie forums where I am pretty active. We don't have a ton of straight GPIO to buttons documentation over there, and I think your setup is a great solution that can effectively bypass controller hardware for a lot of builds.

I might suggest changing the font used in the web interface, as it is a bit hard to read. Also, support for the Mausberry circuit would be fantastic. This is a $16 piece of kit that goes between the USB power input and the wall and provides a physical interface for a power button for the pi. There are several configurations, but you wire up an external switch, and it uses GPIO to trigger a safe shutdown. I could do that with just a button and GPIOneer, but the Mausberry circuit also monitors another pin to cut power to USB and truly turn off the pi when it goes into power/off sleep mode. To make it work, Mausberry suggests a bash script, but this is horribly inefficient so I am using a secondary python script (in addition to GPIOneer). Seems like integration would be a lot better for everyone.

Even without Mausberry integration, I think your setup has a lot of potential and I wish more people would see the virtue in using it.

**Comment by mholgatem** [comment]
cool, good suggestions. I'll be sure to do what I can.

**Comment by mholgatem** [comment]
hey, I'm working on the next iteration of GPioneer, can you send me the bash script that you use for the mausberry circuit?

**Comment by caver01** [comment]
Check here:

https://mausberry-circuits.myshopify.com/pages/setup

**Comment by mholgatem** [comment]
Just an fyi, I finally got around to writing GPIOnext, the successor to GPioneer. I'm just putting the finishing touches on the beta ( still no working web interface ), but it can now emulate 4 joypads as well as keyboard and custom commands.

I looked into implementing the mauseberry-circuit and basically it's just a button and the script just executes the 'sudo poweroff' command. So you will be able to configure a custom command, map it to the gpio that your mauseberry uses, and everything should be good to go.
